waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -140,6 +151,10 @@ proc peers*(pm: PeerManager, proto: string): seq[StoredInfo] =
   # Return the known info for all peers registered on the specified protocol
   pm.peers.filterIt(it.protos.contains(proto))
 
+proc peers*(pm: PeerManager, protocolMatcher: Matcher): seq[StoredInfo] =
+  # Return the known info for all peers matching the provided protocolMatcher
+  pm.peers.filter(proc (storedInfo: StoredInfo): bool = storedInfo.protos.anyIt(protocolMatcher(it)))
```

Comment:
Probably explicit `return` would be better :)

Reply:
Also, a temporary variable to hold the return result would be beneficial for future debugging